src/UserSpaceInstrumentation/AccessTraceesMemoryTest.cpp

```diff
@@ -75,6 +76,8 @@ TEST(AccessTraceesMemoryTest, ReadFailures) {
   pid_t pid = fork();
   CHECK(pid != -1);
   if (pid == 0) {
+    prctl(PR_SET_PDEATHSIG, SIGTERM);
```

Comment:
So far I've only added this call to the child processes in tests that use death tests. But we could consider to put this call into all forked child processes and get rid of the two calls to `kill` and `waitpid` at the end of every test case. This might even be a bit cleaner since `kill` and `waitpid` are only executed when the test succeeds and execution is not interrupted earlier.

Let me know what you think.

Reply:
As discussed offline: I think this is a great idea. I had issues with local runs of the tests where the child processes kept running after termination the test. This would solve that as well.

Reply:
As discussed offline I applied this to all forking tests from the UserSpaceInstrumentation module. PTAL